File: csharp/src/ev1/ut3_strings/ejercicios/resueltos/E04_PrimeraUltima_Resuelto.cs
/*
OBJETIVO: Lee una cadena no vacia y muestra su primer y ultimo caracter.
SOLUCION: ver codigo. Se presenta con menu do-while y opcion 0 para salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

string opcion;

do
{
    Console.WriteLine("=== E04 Primera y ultima letra (resuelto) ===");
    Console.WriteLine("1. Mostrar extremos");
    Console.WriteLine("2. Mostrar extremos y longitud");
    Console.WriteLine("0. Salir");
    Console.Write("Elige una opcion: ");
    opcion = Console.ReadLine() ?? "0";
    Console.WriteLine();

    switch (opcion)
    {
        case "1":
        case "2":
            Console.Write("Escribe una cadena: ");
            string texto = Console.ReadLine() ?? "";

            if (texto.Length > 0)
            {
                Console.WriteLine("Primero: " + texto[0]);
                Console.WriteLine("Ultimo: " + texto[texto.Length - 1]);

                if (opcion == "2")
                {
                    Console.WriteLine("Longitud: " + texto.Length);
                }
            }
            else
            {
                Console.WriteLine("La cadena esta vacia.");
            }
            break;

        case "0":
            Console.WriteLine("Saliendo del ejercicio.");
            break;

        default:
            Console.WriteLine("Opcion no valida.");
            break;
    }

    Console.WriteLine();
} while (opcion != "0");

File: cpp/src/ev1/ut3_strings/ejercicios/resueltos/E04_PrimeraUltima_Resuelto.cpp
/*
OBJETIVO: Lee una cadena no vacia y muestra su primer y ultimo caracter.
SOLUCION: ver codigo. Se presenta con menu do-while y opcion 0 para salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
#include <string>
using namespace std;

int main() {
    string opcion;

    do {
        cout << "=== E04 Primera y ultima letra (resuelto) ===" << endl;
        cout << "1. Mostrar extremos" << endl;
        cout << "2. Mostrar extremos y longitud" << endl;
        cout << "0. Salir" << endl;
        cout << "Elige una opcion: ";
        getline(cin, opcion);
        cout << endl;

        if (opcion == "1" || opcion == "2") {
            cout << "Escribe una cadena: ";
            string texto;
            getline(cin, texto);

            if (texto.size() > 0) {
                cout << "Primero: " << texto[0] << endl;
                cout << "Ultimo: " << texto[texto.size() - 1] << endl;

                if (opcion == "2") {
                    cout << "Longitud: " << texto.size() << endl;
                }
            } else {
                cout << "La cadena esta vacia." << endl;
            }
        } else if (opcion == "0") {
            cout << "Saliendo del ejercicio." << endl;
        } else {
            cout << "Opcion no valida." << endl;
        }

        cout << endl;
    } while (opcion != "0");

    return 0;
}
